src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +573,177 @@ subroutine DustEmission (bounds, &
+            ssr(p) = (K_length+f_0*c_e)/(K_length+c_e)
+
+            ! calculation of the hybrid/total drag partition effect considering both rock and vegetation drag partitioning using LUH2 bare and veg fractions within a grid
+            if (lun%itype(l) == istsoil .or. lun%itype(l) == istcrop) then
+               if (patch%itype(p) == noveg) then
+                  frc_thr_rgh_fct = dpfct_rock(p)
+               else
+                  frc_thr_rgh_fct = ssr(p)
+               end if
             else
-               frc_thr_wet_fct = 1.0_r8
+               frc_thr_rgh_fct = 1.0_r8
             end if
 
-            ! slevis: adding liqfrac here, because related to effects from soil water
+            wnd_frc_slt = fv(p) * frc_thr_rgh_fct   ! wnd_frc_slt will be used in the dust emission equation
 
-            liqfrac = max( 0.0_r8, min( 1.0_r8, h2osoi_liq(c,1) / (h2osoi_ice(c,1)+h2osoi_liq(c,1)+1.0e-6_r8) ) )
+            frc_thr_rghn_fct(p) = frc_thr_rgh_fct   ! save and output hybrid drag partition factor
 
-            ! the following lines come from subr. dst_mbl
-            ! purpose: adjust threshold friction velocity to acct for moisture and
-            !          roughness. The ratio tmp1 / sqrt(forc_rho) comes from
-            !          subr. wnd_frc_thr_slt_get which computes dry threshold
-            !          friction velocity for saltation
+         else
+            wnd_frc_slt = fv(p)                     ! The value here is not important since once lnd_frc_mbl(p) <= 0.0_r8 there will be no emission.
+            frc_thr_rghn_fct(p) = 0.0_r8            ! When LAI > 1, the drag partition effect is zero. dmleung 16 Feb 2024.
+         end if
+
+         !########## end of drag partition effect #######################################################
+
+         ! save soil friction velocity and roughness effect before the if-statement
+         wnd_frc_soil(p) = wnd_frc_slt  ! save soil friction velocity for CLM output, which has drag partition and Owen effect
+         ! save land mobile fraction
+         lnd_frc_mble(p) = lnd_frc_mbl(p)  ! save land mobile fraction first, before the if-statement
+         ! only perform the following calculations if lnd_frc_mbl is non-zero
 
-            wnd_frc_thr_slt = tmp1 / sqrt(forc_rho(c)) * frc_thr_wet_fct * frc_thr_rgh_fct
+         if (lnd_frc_mbl(p) > 0.0_r8) then  ! if bare land fraction is larger than 0 then calculate the dust emission equation
 
             ! reset these variables which will be updated in the following if-block
 
-            wnd_frc_slt = fv(p)
             flx_mss_hrz_slt_ttl = 0.0_r8
             flx_mss_vrt_dst_ttl(p) = 0.0_r8
 
             ! the following line comes from subr. dst_mbl
             ! purpose: threshold saltation wind speed
 
-            wnd_rfr_thr_slt = u10(p) * wnd_frc_thr_slt / fv(p)
+            wnd_rfr_thr_slt = u10(p) * wnd_frc_thr_slt / fv(p)     ! keep and use if we want the default Z03 scheme
```

Comment:
This is only needed for Zender's scheme. It could be taken away from Leung's scheme since Leung does not consider the Owen's effect.